### PR TITLE
Reeti's face tracking system.

### PR DESCRIPTION
**Pull request by mshayganfar**

### CODE REVIEW

**Comment by mshayganfar** [comment]
This face tracking consists of both X and Y direction tracking using neck and eye movements.

File: core/agent/java/edu/wpi/always/client/ReetiFaceTrackerRealizer.java [hunk 1]
```diff
@@ -11,21 +11,37 @@
...
+   private boolean searchFlag = false;
+   
+   private final ReetiPIDMessages reetiPIDOutput;
 
-   private boolean facefoundReeti, falseFaceReeti;
+   private final ReetiCommandSocketConnection client;
+
+   private String lastMessage = "";
 
    public ReetiFaceTrackerRealizer (FaceTrackBehavior params,
```
Comment (charlesrich) [inline core/agent/java/edu/wpi/always/client/ReetiFaceTrackerRealizer.java:51]: Why is this method public?  Always use most limited visibility possible.   Actually, I would not bother with this method, but just make this the definition of run().

File: core/agent/java/edu/wpi/always/client/ReetiPIDController.java [hunk 2]
```diff
@@ -1,167 +1,276 @@
...
-            output = output * SmallTolFactor;
-
-            if ( error > 0 ) // my right
-            {
-               if ( output > 0 )
-                  output = (output + getEyeOut());
-               else
-                  output = ((output * (-1)) + getEyeOut());
+   
+   private double kpXNeck = 0.05, kpYNeck = 0.08; // For the future improvements: kiNeck = 0, kdNeck = 0;
```
Comment (charlesrich) [inline core/agent/java/edu/wpi/always/client/ReetiPIDController.java]: Some of these parameters should be private final static?

File: core/agent/java/edu/wpi/always/client/ReetiPIDController.java [hunk 3]
```diff
@@ -1,167 +1,276 @@
...
+      if((output == 20) || (output == 80))
+         setEyeReachedYLimit(true);
+         
+      setEyeYPIDoutput(output);
    }
-
-   public double getInput () {
-      return Input;
+   
+   public void computeX() {
```
Comment (charlesrich) [inline core/agent/java/edu/wpi/always/client/ReetiPIDController.java]: Why this and computeY public?

**Pull request closed without merging**